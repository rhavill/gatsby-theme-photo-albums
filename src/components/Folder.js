import React from 'react'
import PropTypes from 'prop-types'
import {graphql, Link} from 'gatsby'
import Img from 'gatsby-image'
import {pathToFileTitle} from '../util/text-utils'

const Folder = ({path, imageData}) => {
  const title = pathToFileTitle(path)
  return (
    <article className='folder' data-testid='folder'>
      <Link to={path}>
        <Img fixed={imageData.childImageSharp.fixed} alt={title} title={title} />
        <div className='folder-title'>{title}</div>
      </Link>
    </article>
  )
}

export const query = graphql`
  fragment FolderFragment on File {
    childImageSharp {
      fixed(width: 250, height: 250) {
        ...GatsbyImageSharpFixed
      }
    }
  }
`

Folder.propTypes = {
  path: PropTypes.string.isRequired,
  imageData: PropTypes.shape({
    childImageSharp: PropTypes.shape({
      fixed: PropTypes.shape({
        src: PropTypes.string.isRequired,
        srcSet: PropTypes.string.isRequired,
        width: PropTypes.number.isRequired,
        height: PropTypes.number.isRequired,
      }).isRequired
    }).isRequired  
  }).isRequired
}

export default Folder