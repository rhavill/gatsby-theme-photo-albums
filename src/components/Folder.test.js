import React from "react"
import renderer from "react-test-renderer"
import {cleanup, render} from '@testing-library/react';
import Folder from "./Folder"

// automatically unmount and cleanup DOM after the test is finished.
afterEach(cleanup);

const imageData = {
  childImageSharp: {
    fixed: {
      width: 250,
      height: 250,
      src: '/static/hash/f17d3/folder.png',
      srcSet: '/static/hash/f17d3/folder.png 1x',
    }
  }
}

describe("Folder", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<Folder path='/folder-name' fixedImageData={imageData} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
  it("displays \"title case\" text for folder name", () => {
    const {getByText} = render(
      <Folder path='/first-dir/second-dir/folder-name' fixedImageData={imageData} />
    );
    expect(getByText('Folder Name')).toBeTruthy()
  })
})
