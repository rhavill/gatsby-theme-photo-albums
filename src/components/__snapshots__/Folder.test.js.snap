// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Folder renders correctly 1`] = `
<article
  className="folder"
  data-testid="folder"
>
  <a
    href="/folder-name"
  >
    <div
      className=" gatsby-image-wrapper"
      style={
        Object {
          "display": "inline-block",
          "height": 250,
          "overflow": "hidden",
          "position": "relative",
          "width": 250,
        }
      }
    >
      <picture>
        <source
          srcSet="/static/hash/f17d3/folder.png 1x"
        />
        <img
          alt="Folder Name"
          height={250}
          loading="lazy"
          onLoad={[Function]}
          src="/static/hash/f17d3/folder.png"
          srcSet="/static/hash/f17d3/folder.png 1x"
          style={
            Object {
              "height": "100%",
              "left": 0,
              "objectFit": "cover",
              "objectPosition": "center",
              "opacity": 0,
              "position": "absolute",
              "top": 0,
              "transition": "opacity 500ms",
              "width": "100%",
            }
          }
          width={250}
        />
      </picture>
      <noscript
        dangerouslySetInnerHTML={
          Object {
            "__html": "<picture><source srcset=\\"/static/hash/f17d3/folder.png 1x\\" /><img loading=\\"lazy\\" width=\\"250\\" height=\\"250\\" srcset=\\"/static/hash/f17d3/folder.png 1x\\" src=\\"/static/hash/f17d3/folder.png\\" alt=\\"Folder Name\\" style=\\"position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
          }
        }
      />
    </div>
    <div
      className="folder-title"
    >
      Folder Name
    </div>
  </a>
</article>
`;
